core/templates/pages/contributor-dashboard-page/modal-templates/question-suggestion-review-modal.controller.ts

```diff
@@ -102,38 +86,126 @@ angular.module('oppia').controller('QuestionSuggestionReviewModalController', [
       });
     };
 
-    $scope.init = function() {
+    $scope.reviewable = reviewable;
+    $scope.misconceptionsBySkill = misconceptionsBySkill;
+    $scope.currentSuggestionId = suggestionId;
+
+    $scope.currentSuggestion = suggestionIdToContribution[suggestionId];
+    delete suggestionIdToContribution[suggestionId];
+    $scope.remainingContributionIds = Object.keys(
+      suggestionIdToContribution
+    );
+    $scope.remainingContributionIds.reverse();
+    $scope.skippedContributionIds = [];
+    $scope.allContributions = suggestionIdToContribution;
+    $scope.allContributions[suggestionId] = $scope.currentSuggestion;
+
+    $scope.isLastItem = $scope.remainingContributionIds.length === 0;
+    $scope.isFirstItem = $scope.skippedContributionIds.length === 0;
+
+    $scope.refreshContributionState = function() {
+      $scope.suggestion = (
+        $scope.allContributions[$scope.currentSuggestionId].suggestion);
+      $scope.question = QuestionObjectFactory.createFromBackendDict(
+        $scope.suggestion.change.question_dict);
+      $scope.authorName = $scope.suggestion.author_name;
+      $scope.contentHtml = $scope.question.getStateData().content.html;
+      $scope.questionHeader = (
+        $scope.allContributions[
+          $scope.currentSuggestionId].details.skill_description);
+      $scope.skillRubrics = (
+        $scope.allContributions[
+          $scope.currentSuggestionId].details.skill_rubrics);
+      $scope.questionStateData = $scope.question.getStateData();
+      $scope.questionId = $scope.question.getId();
+      $scope.canEditQuestion = false;
+      $scope.skillDifficulty = $scope.suggestion.change.skill_difficulty;
+      $scope.skillDifficultyLabel = getSkillDifficultyLabel();
+      $scope.skillRubricExplanations = getRubricExplanation(
+        $scope.skillDifficultyLabel);
+      $scope.reviewMessage = '';
+      $scope.suggestionIsRejected = $scope.suggestion.status === 'rejected';
+
       if (reviewable) {
         SiteAnalyticsService
           .registerContributorDashboardViewSuggestionForReview('Question');
       } else if ($scope.suggestionIsRejected) {
-        _getThreadMessagesAsync(suggestionId);
+        _getThreadMessagesAsync($scope.currentSuggestionId);
       }
+      $scope.showQuestion = true;
+      $rootScope.$applyAsync();
     };
 
-    $scope.init();
+    $scope.refreshContributionState();
 
     $scope.questionChanged = function() {
       $scope.validationError = null;
     };
 
+    $scope.refreshActiveContributionState = function() {
+      let nextContribution = $scope.allContributions[
```

Comment:
Prefer `const` if the variable is not being reassigned.

Reply:
Done